apps/tokenization/src/agreement/agreement.controller.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Logger,
+  Param,
+  Post,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import {
+  ApiBody,
+  ApiCreatedResponse,
+  ApiOkResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { AgreementDTO, AgreementService, BlockchainPropertiesService, CreateAgreementDTO } from '@zero-labs/tokenization-api';
+import { providers, Wallet } from 'ethers';
+import { signAgreement } from '@zero-labs/tokenization';
+import { AccountService } from '../account/account.service';
+
+@ApiTags('agreement')
+@Controller('agreement')
+@UsePipes(ValidationPipe)
+export class AgreementController {
+  private readonly logger = new Logger(AgreementController.name, {
+    timestamp: true,
+  });
+
+  constructor(
+    private readonly blockchainPropertiesService: BlockchainPropertiesService,
+    private readonly agreementService: AgreementService,
+    private readonly accountService: AccountService
+  ) {}
+
+  @Post()
+  @ApiBody({ type: CreateAgreementDTO })
+  @ApiCreatedResponse({
+    type: AgreementDTO,
+    description: 'Agreement data',
+  })
+  public async create(
+    @Body() params: CreateAgreementDTO,
+  ): Promise<AgreementDTO> {
+    const agreement = await this.agreementService.create(params);
+    const seller = await this.accountService.get(params.seller);
+    const buyer = await this.accountService.get(params.buyer);
+
+    const { rpcNode } = await this.blockchainPropertiesService.get();
+    const provider = new providers.JsonRpcProvider(rpcNode);
+
+    const buyerWallet = new Wallet(buyer.privateKey, provider);
+    const sellerWallet = new Wallet(seller.privateKey, provider);
```

Comment:
are those keys stored securely?

Reply:
We have a task for that - just haven't gotten around to it.
Will be a different PR.